tests/suites/test_suite_psa_crypto.data

```diff
@@ -5080,6 +5080,22 @@ PSA key derivation: ECDH on P256 with HKDF-SHA256, missing info
 depends_on:PSA_WANT_ALG_ECDH:PSA_WANT_ALG_HKDF:PSA_WANT_ALG_SHA_256:PSA_WANT_KEY_TYPE_ECC_KEY_PAIR:MBEDTLS_PK_PARSE_C:PSA_WANT_ECC_SECP_R1_256
 derive_input:PSA_ALG_KEY_AGREEMENT(PSA_ALG_ECDH, PSA_ALG_HKDF(PSA_ALG_SHA_256)):PSA_KEY_DERIVATION_INPUT_SALT:PSA_KEY_TYPE_NONE:"":PSA_SUCCESS:PSA_KEY_DERIVATION_INPUT_SECRET:PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1):"c88f01f510d9ac3f70a292daa2316de544e9aab8afe84049c62a9c57862d1433":PSA_SUCCESS:0:UNUSED:"":UNUSED:PSA_KEY_TYPE_NONE:PSA_ERROR_BAD_STATE
 
+PSA key derivation: TLS12_ECJPAKE_TO_PMS, good input, output too short
```

Comment:
Lots more missing test cases here: positive and negative test cases with correct inputs and varying output lengths, negative test cases with different inputs.

Reply:
I would assume that this is tested one layer deeper, here: https://github.com/Mbed-TLS/mbedtls/pull/6115/files#diff-cc82e7d511f22a0a29b24a8ad0c62d6c62b5f8c84ad67e55ea71d7fc035101aeR5800-R5840
But sure, I guess that these tests don't take long enough for duplication to be a problem and checking things on a different layer is also fine.

Reply:
@gilles-peskine-arm What do you want to do about the tests? It might be a bit confusing to have a partial set here and a complete set elsewhere (the place pointed by Andrzej). Probably better to either have only one complete set if possible, or two complete sets if there's a reason to have them in both places.

Reply:
I initially added those positive cases because I wanted negative test cases when the input comes from a key agreement. But then I realized there was a test case with key agreement chaining in the op-fail test suite. Looking back I'm not completely happy with relying on that because it's a corner case in the test code generator and there's a risk we'll stop generating the test case and not really notice. So I think I should add a manual test here too.

Also, I had not realized that `derive_ecjpake_to_pms` was testing the same thing. I find it weird that this KDF has its own function. What does it do that's missing in the generic `derive` functions?

Reply:
> Also, I had not realized that `derive_ecjpake_to_pms` was testing the same thing. I find it weird that this KDF has its own function. What does it do that's missing in the generic `derive` functions?

I'm not sure. It's possible that we were just not (sufficiently) aware of the existence of the generic `derive` functions. If that's the case, then we could just port the existing test cases to the generic function.